nav2_constrained_smoother/include/nav2_constrained_smoother/smoother_cost_function.hpp

```diff
@@ -159,7 +159,7 @@ class SmootherCostFunction
     Eigen::Matrix<T, 2, 1> center = arcCenter(
       pt_prev, pt, pt_next,
       next_to_last_length_ratio_ < 0);
-    if (ceres::isinf(center[0])) {
+    if (isInfinite(center[0])) {
```

Comment:
Why not just `ifdef` on the version number of ceres? 

By the way, Tier 3 means that things might not really work but we generally try to make it work if at all possible, but I don't think we need to worry about it too much as long as 20.04's distribution's API are correct in their respective branches

Reply:
> Why not just ifdef on the version number of ceres?

I relied mainly on Ceres official docs where they are recommending to use `std::isinf()` functions for such comparisons. Although, the solution with `#ifdef` will also be fine, I think. Something like:
```
#if defined(USE_OLD_CERES_API)
  #define CERES_ISINF(x) ceres::IsInfinite(x)
#else
  #define CERES_ISINF(x) ceres::isinf(x)
#endif
```

Reply:
I don’t mind much one way or another, that’s just how we deal with such problems in PCL so its familiar to me. If you prefer this, that works too. 

Can you confirm for me this builds given our CI issues?

Reply:
Okay, I finally decided to use latest solution with Ceres versions defined in compile-time, as more straightforward and elegant (we will re-use existing API, instead of making something again) solution. Force-pushed the commit.

I could confirm that the build works with:
* Tier-III system
  * Ubuntu 20.04
  * ROS2 Rolling built from sources (2023-03-10)
  * Ceres v1.14.0-4ubuntu1.1
  * Nav2 build ... OK
* navigation2:main Docker container
  * Ubuntu 22.04
  * ROS2 Rolling installed from binaries ~2023.01.31
  * Ceres v2.0.0+dfsg1-5
  * Nav2 build ... OK
* navigation2:main Docker container w/ latest Ceres (pre-2.2.0)
  * Ubuntu 22.04
  * ROS2 Rolling installed from binaries ~2023.01.31
  * Ceres was replaced to built one from 2023.03.14 sources
  * Nav2 build ... OK